pyproject.toml

```diff
@@ -128,15 +128,23 @@ ignore = "W009"
 src = ["src"]
 line-length = 88
 select = [
+    "B",  # bugbear
     "D",  # pydocstyle
     "E",  # pycodestyle
     "F",  # pyflakes
     "I",  # isort
+    "PYI", # flake8-pyi
     "UP", # pyupgrade
     "RUF", # ruff
     "W",  # pycodestyle
 ]
 ignore = [
+    # bugbear ignore
+    "B004", # Using `hasattr(x, "__call__")` to test if x is callable is unreliable.
+    "B007", # Loop control variable `i` not used within loop body
+    "B009", # Do not call `getattr` with a constant attribute value
+    "B010", # [*] Do not call `setattr` with a constant attribute value.
```

Comment:
Would remove the `[*]`

Reply:
👍 About this one, getattr and hasattr(x. "__call__"): there's probably something to do (and an autofix) but then mypy is unhappy. It seems like a big design change and I didn't know what was best. Interesting warning, but hard to fix (for me at least).

Reply:
I think it's fine to ignore whatever's causing too much trouble, can always unignore later.